library/core/src/main/java/com/google/android/exoplayer2/mediacodec/MediaCodecRenderer.java

```diff
@@ -2128,6 +2130,12 @@ private boolean isDecodeOnlyBuffer(long presentationTimeUs) {
         return true;
       }
     }
+    if (codecNeedsSkipSurplusOutBufferWorkaround && size > 0) {
+      if (Math.min(decodeOnlyPresentationTimestamps.get(0), decodeOnlyPresentationTimestamps.get(size - 1)) < presentationTimeUs
+          && Math.max(decodeOnlyPresentationTimestamps.get(0), decodeOnlyPresentationTimestamps.get(size - 1)) > presentationTimeUs) {
+        return true;
+      }
```

Comment:
After modifying the code to:

1. add the SEEK_TO intent (sample code in the bug)
2. so the `codecNeedsSkipSurplusOutBufferWorkaround` is set when switching to 1080p

The return true on line 2136 is never executed.

Please follow the test steps in the bug exactly and add some logging to prove this code actually works, as @tonihei  noted this is a path that Google team is contemplating in the future, but for now it would be much preferred to disable the Auto-FRC if possible rather than complicating this code (which is in the critical render path).

Thanks!